Sources/eppo/UniversalFlagConfig.swift

```diff
@@ -0,0 +1,145 @@
+    // todo: add bandits
+    
+    static func decodeFromJSON(from jsonString: String) throws -> UniversalFlagConfig {
+        guard let jsonData = jsonString.data(using: .utf8) else {
+            throw UniversalFlagConfigError.notUTF8Encoded("Failed to encode JSON string into UTF-8 data.")
+        }
+
+        let decoder = JSONDecoder()
+        
+        // Set up the date formatter
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZ"  // Adjusted to include milliseconds
+        dateFormatter.locale = Locale(identifier: "en_US_POSIX")  // Use POSIX to avoid unexpected behaviors
+        dateFormatter.timeZone = TimeZone(secondsFromGMT: 0)  // Adjust if your JSON dates are not in GMT
+        
+        // Use the date formatter in the decoder
+        decoder.dateDecodingStrategy = .formatted(dateFormatter)
+        
+        do {
+            return try decoder.decode(UniversalFlagConfig.self, from: jsonData)
+        } catch let error as DecodingError {
+            switch error {
+            case .typeMismatch(_, let context):
+                throw UniversalFlagConfigError.parsingError("Type mismatch: \(context.debugDescription)")
+            case .valueNotFound(_, let context):
+                throw UniversalFlagConfigError.parsingError("Value not found: \(context.debugDescription)")
+            case .keyNotFound(let key, let context):
+                throw UniversalFlagConfigError.parsingError("Key not found: \(key.stringValue) - \(context.debugDescription)")
+            case .dataCorrupted(let context):
+                throw UniversalFlagConfigError.parsingError("Data corrupted: \(context.debugDescription)")
+            default:
+                throw UniversalFlagConfigError.parsingError("JSON parsing error: \(error.localizedDescription)")
+            }
+        } catch {
+            throw UniversalFlagConfigError.parsingError("Unexpected error: \(error.localizedDescription)")
+        }
+    }
+}
+
+// enums
+enum UniversalFlagConfigError: Error, CustomNSError, LocalizedError {
+    case notUTF8Encoded(String)
+    case parsingError(String)
+
+    static var errorDomain: String { return "UniversalFlagConfigError" }
+    
+    var errorCode: Int {
+        switch self {
+        case .notUTF8Encoded:
+            return 100 
+        case .parsingError:
+            return 101
+        }
+    }
+
+    var errorDescription: String? {
+        switch self {
+        case .notUTF8Encoded(let message):
+            return message
+        case .parsingError(let message):
+            return message
+        }
+    }
+}
+
+enum UFC_VariationType: String, Decodable {
+      case boolean = "BOOLEAN"
+      case integer = "INTEGER"
+      case json = "JSON"
+      case numeric = "NUMERIC"
+      case string = "STRING"    
+}
+
+enum UFC_AlgorithmType: String, Decodable {
+    case constant = "CONSTANT"
+    case contextualBandit = "CONTEXTUAL_BANDIT"
+}
```

Comment:
This is not part of the UFC so we should not need this type

Reply:
I got it from the UFC API interface definition. Are you saying that interface is wrong or this is out of scope of the UFC. If the latter yes I know but it doesn't hurt to parse it.

Reply:
It's not part of the config, see https://github.com/Eppo-exp/sdk-test-data/blob/main/ufc/flags-v1.json for the canonical UFC config definition